File: rebus-rebuscrus/src/java/it/csi/rebus/rebuscrus/business/service/ZipService.java
/*
* SPDX-FileCopyrightText: (C) Copyright 2022 Regione Piemonte
* SPDX-License-Identifier: EUPL-1.2
*/
package it.csi.rebus.rebuscrus.business.service;

import it.csi.rebus.rebuscrus.vo.FiltroContribuzioneVO;

public interface ZipService {

	public byte[] zipRicercaContribuzione(FiltroContribuzioneVO filtro);
}

File: rebus-rebuscruswcl/src/app/services/zip.service.ts
/*************************************************
Copyright Regione Piemonte - 2022
SPDX-License-Identifier: EUPL-1.2-or-later
***************************************************/

import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { map } from 'rxjs/operators';

@Injectable()
export class ZipService {

    constructor(
        private http: HttpClient
    ) { }



    zipRicercaContribuzione(body: any) {
        var url: string = "/parcobustplweb/restfacade/zip/zipRicercaContribuzione";
        return this.http.post(url, body, { responseType: 'blob' }).pipe(map(
            (res) => {
                return new Blob([res], { type: 'application/zip' })
            }
        ));
    }
}